src/index.ts

```diff
@@ -154,6 +156,47 @@ function shouldSendFlowLogToSplunk(msg: string): boolean {
   return action === 'ACCEPT' && !rejectedProtocols.includes(protocol)
 }
 
+function getLogTimeStampRegex(logType: CloudWatchLogTypes): RegExp | undefined {
+  switch (logType) {
+    case CloudWatchLogTypes.app:
+      return /"@timestamp"\s*:\s*"(.*?)"/
+    case CloudWatchLogTypes.squid:
+      return /^(\d+\.\d{3})/
+    case CloudWatchLogTypes.syslog:
+      return /^(.+?)\s*ip/
+    default:
+      console.log(`Time stamp parsing not yet implemented for "${logType}" log types.`)
+      return undefined
+  }
+}
+
+// TODO: whilst adding this in it'll return undefined if not yet implemented for log type.
+function parseTimeFromLog(log: string, logType: CloudWatchLogTypes): number | undefined {
+  const timeStampRegex = getLogTimeStampRegex(logType)
+  if (timeStampRegex === undefined) {
+    return undefined
+  }
+
+  const logTimeMatches = log.match(timeStampRegex)
+  console.log(logTimeMatches)
+  if (logTimeMatches === null || logTimeMatches.length < 1) {
+    console.error(`Failed to parse time stamp from log of type "${logType}" using "${timeStampRegex}"`)
+    return undefined
+  }
+
+  // Squid Access Log time format is already in the correct format for Splunk.
+  if (logType === CloudWatchLogTypes['squid']) {
+    return Number(logTimeMatches[1])
+  }
+
+  const epoch = Date.parse(logTimeMatches[1])
```

Comment:
Cool that this works for the syslog format :)

Reply:
It does, though there is a disclaimer on the Date.parse docs about how it handles [non-standard strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse#non-standard_date_strings) (and there's no argument to pass a string format). I thought I'd see how it goes before looking at some other library since I'm sure we'd like to keep dependencies to an absolutely minimum...

Reply:
After looking a bit closer... it doesn't since without the year it does something rather unexpected in not using the current year :) ... back to the drawing board.
```
> new Date('Feb 10 10:16:36')
2001-02-10T10:16:36.000Z
```